scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,385 @@
+        local spell = GetSpell(spellID)
+        if spell then
+            if spell:getSkillType() == xi.skill.ENHANCING_MAGIC then
+                -- half casting delay
+                pet:setLocalVar(buffModeVar, 0)
+            end
+-- print(spellTarget:getName() .. '->' .. spellID)
+            pet:castSpell(spellID, spellTarget or pet)
+            setMagicCastCooldown(pet)
+            return -1
+        end
+    end
+end
+
+xi.pets.avatar.getSpiritSpell = function(pet)
+    local spellID = 0
+    local spellTarget = nil
+    local petID = pet:getPetID()
+    if petID == xi.petId.LIGHT_SPIRIT then
+        spellID, spellTarget = xi.pets.avatar.getLightSpiritSpell(pet)
+    -- elseif not pet:isEngaged() then
+    --     -- to show how we can further tune idle casting
+    --     if not pet:hasStatusEffect(xi.effect.SHOCK_SPIKES) then
+    --         spellID = xi.magic.spell.SHOCK_SPIKES
+    --         spellTarget = pet
+    --     end
+    end
+
+    return spellID, spellTarget
+end
+
+-- Disable cyclomatic complexity check for this function:
+-- luacheck: ignore 561
+xi.pets.avatar.getLightSpiritSpell = function(pet)
+    -- returns the spirit's preferred target based on positioning
+    local master = pet:getMaster()
+    local posTarget = nil
+    local posSpellID = 0
+    local buffTarget = nil
+    local buffSpellID = 0
+    local cureTarget = nil
+    local spellID = 0
+
+    if master then
+        if not pet:isEngaged() then
+            pet:lookAt(master:getPos())
+        else
+            -- don't do logic to buff anyone if pet is engaged. buffTarget will be reset below
+            buffTarget = master
+        end
+
+        local distance  = pet:checkDistance(master) -- starts as distance to master, updated to be distance to the posTarget
+        local hpp       = 0
+        local alliance  = master:getAlliance()
+        local party     = master:getParty()
+        local petLvl    = pet:getMainLvl()
+
+        for _, member in pairs(alliance) do
+            -- tiered chance of healing based on how low they are
+            local tempHPP = member:getHPP()
+            if
+                pet:checkDistance(member) < 20 and
+                (hpp == 0 or
+                tempHPP < hpp) and
+                ((tempHPP < 75 and
+                math.random(100) < 50) or
+                (tempHPP < 50 and
+                math.random(100) < 50) or
+                (tempHPP < 25 and
+                math.random(100) < 50))
+            then
+                hpp = tempHPP
+                cureTarget = member
+            end
+        end
+
+        if cureTarget then
+            -- curaga can be used on anyone in the alliance and isn't determined by how many need a heal (assumed to be based on skill over cap)
+            if
+                petLvl >= 16 and
+                math.random(100) <= 2 * xi.summon.getSummoningSkillOverCap(pet)
+            then
+                -- curaga
+                if petLvl >= 71 then
```

Comment:
This block looks suspiciously like it can be written as a table+driver

Reply:
Done. Hopefully you like it. I'm pretty happy of the results